src/Dual/DualCategory.lidr

```diff
@@ -0,0 +1,78 @@
+\iffalse
+SPDX-License-Identifier: AGPL-3.0-only
+
+This file is part of `idris-ct` Category Theory in Idris library.
+
+Copyright (C) 2019 Stichting Statebox <https://statebox.nl>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+\fi
+
+> module Dual.DualCategory
+>
+> import Basic.Category
+>
+> %access public export
+> %default total
+>
+> dualMorphism:
+>      (cat : Category)
+>   -> (a, b  : obj cat)
+>   -> Type
+> dualMorphism cat a b = mor cat b a
+>
+> dualCompose :
+>      (cat : Category)
+>   -> (a, b, c : obj cat)
+>   -> (g : mor cat b a)
```

Comment:
what do you say about keeping the `f` and `g` order in the type signature and reversing it (first `g`, then `f`) in the implementation?

Moreover, maybe it would be nice to write directly `f : dualMorphism cat a b`